lib/matplotlib/style/core.py

```diff
@@ -66,22 +67,92 @@ def use(style):
     else:
         styles = style
 
+    styles = insert_parent_styles(styles)
+
     for style in styles:
         if not cbook.is_string_like(style):
             mpl.rcParams.update(style)
             continue
 
-        if style in library:
-            mpl.rcParams.update(library[style])
+        mpl.rcParams.update(get_style_parameters(style))
+
+
+def insert_parent_styles(styles):
+    """Returns a list with all the styles and parents styles called"""
+    tmp_styles = []
+    for style in styles:
+        style = _get_parent_styles_from_str_or_dict(style)
+        tmp_styles.extend(style)
+    return tmp_styles[:]
+
+
+def _get_parent_styles_from_str_or_dict(styles):
+    """Returns the parent styles from a single style or a dict"""
+    styles = [styles]
+    has_parent = True
+    visited_styles = []
+    temp_styles = styles[:]
+
+    while has_parent:
+        for style in styles:
+            if style in visited_styles:
+                continue
+            visited_styles.append(style)
+
+            if _is_dict_like(style):
+                style_dic = style
+            else:
+                style_dic = get_style_parameters(style)
+
+            has_parent = 'style' in style_dic
+            if has_parent:
+                parent = style_dic['style']
+                temp_styles.insert(temp_styles.index(style)+1, parent)
+                temp_styles = list(_list_flatten(temp_styles))
+
+        # Style update before scanning again for parents
+        styles = temp_styles[:]
+    styles = styles[::-1]
+    return styles
+
+
+def _is_dict_like(style):
+    """returns whether the element looks like a dict, meaning it is a dict
+    or an OrderedDict."""
+    if type(style) is dict:
+        return True
+    if sys.version_info[:2] >= (2, 7):
+        from collections import OrderedDict
+        if type(style) is OrderedDict:
+            return True
+
+
+def get_style_parameters(style):
+    """Get the parameter rc of the style, whether it is a file or a
+    dictionnary"""
```

Comment:
I don't mean to be repetitive, but you might have missed my earlier comment: This function doesn't handle dictionaries, so this docstring is misleading.